bluesky-social/social-app
# Upgrade Reanimated from 2.14.4 to 3.1.0

## Summary 
Right now, Bluesky app uses Reanimated of version 2.14.4. This PR introduces an update for this library to 3.1.0 version.
It also removes the need to use `useLegacyImplementation` in drawer, introduced in PR #585, as in Reanimated version 3.1.0 drawer works properly on Android without using old implementation.

Fixes #751.

## Test plan
1. Run the AT Protocol server and mobile app using dev environment.
2. Create new account using launched locally AT Proto server as the hosting provider.

### Testing modals
As react-native-bottom-sheet uses react-native-reanimated behind the hood, it's worth testing if it still works on newer version of Reanimated.

3. Create a post with any content.
4. Try to repost it and then delete it.
Bottom sheet should work as usual.

https://github.com/bluesky-social/social-app/assets/23281839/30e9b7f6-9926-4a35-b7e9-a3bbf9a384c8

### Testing drawer
As this PR removes using `useLegacyImplementation`, it's worth testing if drawer works without this feature.

3. From the homepage open the drawer.
4. Try to open most of the options, available from it.
5. Try to close the drawer.

https://github.com/bluesky-social/social-app/assets/23281839/e06b8057-7676-4bdf-9e57-b4d957552f09
